src/electron/contextMenu.ts

```diff
@@ -1,9 +1,57 @@
-import { Menu } from "electron";
-
-export const textEditContextMenu = Menu.buildFromTemplate([
-  { label: "切り取り", role: "cut" },
-  { label: "コピー", role: "copy" },
-  { label: "貼り付け", role: "paste" },
-  { type: "separator" },
-  { label: "全選択", role: "selectAll" },
-]);
+import { Menu, MenuItem, MenuItemConstructorOptions } from "electron";
+import { ContextMenuType, TextEditContextMenuAction } from "@/type/contextMenu";
+
+type ContextMenuItemConstructorOptions<T> = MenuItemConstructorOptions & {
+  label?: T;
+};
+
+const defaultMenuItemOptions: Record<
+  ContextMenuType,
+  MenuItemConstructorOptions[]
+> = {
+  TEXT_EDIT: [
+    {
+      label: "切り取り",
+      role: "cut",
+    },
+    {
+      label: "コピー",
+      role: "copy",
+    },
+    {
+      label: "貼り付け",
+      role: "paste",
+    },
+    {
+      type: "separator",
+    },
+    {
+      label: "全選択",
+      role: "selectAll",
+    },
+  ] as ContextMenuItemConstructorOptions<TextEditContextMenuAction>[],
+};
+
+// 右クリックする場所(TEXT_EDITなど)ごとに動作を保持する仕様にしましたが、
+// 最後にクリックした場所だけを保存でも良いかもしれません
+const menus = new Map<ContextMenuType, Menu>();
+
+// 返り値の型は Electron.Menu | undefined ではなく Electron.Menu 確定のはずですがどうすればいいか分かりません
+const checkOrBuildContextMenu = (type: ContextMenuType) => {
+  if (menus.has(type)) return menus.get(type);
```

Comment:
まあhasがtrueならgetがundefinedにならないのはプログラマ的には明確なので、こうとかでしょうか。
```suggestion
const checkOrBuildContextMenu = (type: ContextMenuType) => {
  if (menus.has(type)) {
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    return menus.get(type)!;
  }
```
あるいはこうとか！
```suggestion
const checkOrBuildContextMenu = (type: ContextMenuType) => {
  if (menus.has(type)) {
    const menu = menus.get(type);
    if (!menu) throw new Error("menu is undefined");
    return menu;
  }
```